src/library/scala/collection/mutable/FlatHashTable.scala

```diff
@@ -219,7 +219,9 @@ trait FlatHashTable[A] extends FlatHashTable.HashUtils[A] {
     tableSize = 0
     nnSizeMapReset(table.length)
     seedvalue = tableSizeSeed
-    threshold = newThreshold(_loadFactor, table.length)
+    // If the table length reaches MaximumTableCapacity of Int.MaxValue / 2 then set the
+    // threshold to Int.MaxValue - 8 as a safe value
+    threshold = if (table.length >= MaximumTableCapacity) Int.MaxValue - 8 else newThreshold(_loadFactor, table.length)
```

Comment:
Isn't it too late at this point? This is only the threshold. The new table has already been allocated a few lines earlier (or not, if `table.length * 2` overflows). The same goes for `HashTable` as far as I can tell. Only `OpenHashMap` limits the actual size.

Reply:
Ah, yes, you're right.

Reply:
I was just looking again at the possibility of already being over the size limit before recomputing the new threshold but it doesn't seem like this is possible. There's no way to instantiate the class with a larger initial size. Nevertheless I think it would be cleaner to do this computation inside of `newThreshold`.